File: portifolio-api/src/main/java/com/portifolio/backend/model/Skill.java
package com.portifolio.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "skills")
@Data 
@Builder 
@NoArgsConstructor 
@AllArgsConstructor
public class Skill {
    
    @Id 
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 80)
    private String name;

    @Column(nullable = false, length = 60)
    private String category;

    @Column(nullable = false)
    private Integer proficiency;

    @Column(name = "icon_name", length = 80)
    private String iconName;

    @Builder.Default
    @Column(name = "sort_order")
    private Integer sortOrder = 0;

    @Builder.Default
    @Column(nullable = false)
    private Boolean active = true;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist 
    protected void onCreate() { 
        createdAt = LocalDateTime.now(); 
    }
}

File: frontend/src/types/skill.ts
export interface Skill {
  id: number;
  name: string;
  category?: string;
  proficiency: number; // 0-100
  displayOrder: number;
  createdAt: string;
  updatedAt: string;
}

export interface SkillPayload {
  name: string;
  category?: string;
  proficiency: number;
  displayOrder: number;
}
